In [1]:
%load_ext autoreload
%autoreload 2

In [2]:
import pandas as pd

#  📊 Raw Data Exploration | Meat Consumption Project

**Description:** 
In this notebook, we load and explore the raw datasets related to global meat consumption. The goal is to understand the structure, completeness, and key variables of each dataset. We also identify any necessary preprocessing steps and visualize the available data to guide the next stages of analysis.

--

**Structure:**
- [Loading raw data](#loading-raw-data)
- [FAO Dataset](#fao-dataset)
- [GDP Dataset](#gdp-dataset)
- [Urban Dataset](#urban-dataset)
- [Edudcation Dataset](#education-dataset)
- [Enviroment Dataset](#enviroment-dataset)
- [Production Dataset](#meat-production-dataset)

--

## Loading raw data

In [25]:
# file dictionary
files = {
    'faostat': '../Data/raw/faostat_meat_kg_per_capita.csv',
    'gdp': '../Data/raw/worldbank_gdp_per_capita.csv',
    'urban':'../Data/raw/worldbank_urban_population.csv',
    'education':'../Data/raw/share-of-the-world-population-with-at-least-basic-education.csv',
    'enviroment':'../Data/raw/food_enviroment_impact.csv',
    'production':'../Data/raw/global-meat-production.csv'
}

#l configuration dictionary
config = {
    'faostat': {},
    'gdp': {'header':2},
    'urban':{'header':2},
    'education':{},
    'enviroment':{},
    'production':{}
    }
# frame work dictinary
dfs = {

    key: pd.read_csv(path,**config.get(key, {}))
    for key, path in files.items()
}

## FAO Dataset

**Description.**
This dataset provides the per capita meat consumption (in kg/year) for different countries and types of meat.


In [26]:
# First we simplify the name of the frame we will explore. 
faostat = dfs['faostat']

In [5]:
# File header structure
faostat.head(3)

,Domain Code,Domain,Area Code (M49),Area,Element Code,Element,Item Code (FBS),Item,Year Code,Year,Unit,Value,Flag,Flag Description,Note
0,FBS,Food Balances (2010-),4,Afghanistan,645,Food supply quantity (kg/capita/yr),S2731,Bovine Meat,2010,2010,kg/cap,4.74,E,Estimated value,NaN
1,FBS,Food Balances (2010-),4,Afghanistan,645,Food supply quantity (kg/capita/yr),S2731,Bovine Meat,2011,2011,kg/cap,4.80,E,Estimated value,NaN
2,FBS,Food Balances (2010-),4,Afghanistan,645,Food supply quantity (kg/capita/yr),S2731,Bovine Meat,2012,2012,kg/cap,4.40,E,Estimated value,NaN


In [6]:
# file tail structure
faostat.tail(3)

,Domain Code,Domain,Area Code (M49),Area,Element Code,Element,Item Code (FBS),Item,Year Code,Year,Unit,Value,Flag,Flag Description,Note
11862,FBS,Food Balances (2010-),716,Zimbabwe,645,Food supply quantity (kg/capita/yr),S2735,"Meat, Other",2020,2020,kg/cap,2.35,E,Estimated value,NaN
11863,FBS,Food Balances (2010-),716,Zimbabwe,645,Food supply quantity (kg/capita/yr),S2735,"Meat, Other",2021,2021,kg/cap,2.35,E,Estimated value,NaN
11864,FBS,Food Balances (2010-),716,Zimbabwe,645,Food supply quantity (kg/capita/yr),S2735,"Meat, Other",2022,2022,kg/cap,2.35,E,Estimated value,NaN


In [7]:
faostat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11865 entries, 0 to 11864
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Domain Code       11865 non-null  object 
 1   Domain            11865 non-null  object 
 2   Area Code (M49)   11865 non-null  int64  
 3   Area              11865 non-null  object 
 4   Element Code      11865 non-null  int64  
 5   Element           11865 non-null  object 
 6   Item Code (FBS)   11865 non-null  object 
 7   Item              11865 non-null  object 
 8   Year Code         11865 non-null  int64  
 9   Year              11865 non-null  int64  
 10  Unit              11865 non-null  object 
 11  Value             11865 non-null  float64
 12  Flag              11865 non-null  object 
 13  Flag Description  11865 non-null  object 
 14  Note              0 non-null      float64
dtypes: float64(2), int64(4), object(9)
memory usage: 1.4+ MB


In [8]:
faostat.isnull().sum()

Domain Code             0
Domain                  0
Area Code (M49)         0
Area                    0
Element Code            0
Element                 0
Item Code (FBS)         0
Item                    0
Year Code               0
Year                    0
Unit                    0
Value                   0
Flag                    0
Flag Description        0
Note                11865
dtype: int64

In [9]:
print('(rows, columns):',faostat.shape)
print('columns:', faostat.columns)

(rows, columns): (11865, 15)
columns: Index(['Domain Code', 'Domain', 'Area Code (M49)', 'Area', 'Element Code',
       'Element', 'Item Code (FBS)', 'Item', 'Year Code', 'Year', 'Unit',
       'Value', 'Flag', 'Flag Description', 'Note'],
      dtype='object')


In [11]:
# General exploration, number of data and features
print('Unic values in  columns')
print(faostat['Domain'].unique())
print(faostat['Item Code (FBS)'].unique())
print(faostat['Item'].unique())
print('Countries:', faostat['Area'].nunique())
print('Year range:',faostat['Year'].min(), '-' , faostat['Year'].max())
print('Suspicious values',faostat['Flag'].value_counts())

Unic values in  columns
['Food Balances (2010-)']
['S2731' 'S2732' 'S2733' 'S2734' 'S2735']
['Bovine Meat' 'Mutton & Goat Meat' 'Pigmeat' 'Poultry Meat' 'Meat, Other']
Countries: 190
Year range: 2010 - 2022
Suspicious values Flag
E    11865
Name: count, dtype: int64


In [41]:
print(faostat['Area'].unique())

['Afghanistan' 'Albania' 'Algeria' 'Angola' 'Antigua and Barbuda'
 'Argentina' 'Armenia' 'Australia' 'Austria' 'Azerbaijan' 'Bahamas'
 'Bahrain' 'Bangladesh' 'Barbados' 'Belarus' 'Belgium' 'Belize' 'Benin'
 'Bhutan' 'Bolivia (Plurinational State of)' 'Bosnia and Herzegovina'
 'Botswana' 'Brazil' 'Bulgaria' 'Burkina Faso' 'Burundi' 'Cabo Verde'
 'Cambodia' 'Cameroon' 'Canada' 'Central African Republic' 'Chad' 'Chile'
 'China' 'China, Hong Kong SAR' 'China, Macao SAR' 'China, mainland'
 'China, Taiwan Province of' 'Colombia' 'Comoros' 'Congo' 'Costa Rica'
 "Côte d'Ivoire" 'Croatia' 'Cuba' 'Cyprus' 'Czechia'
 "Democratic People's Republic of Korea"
 'Democratic Republic of the Congo' 'Denmark' 'Djibouti' 'Dominica'
 'Dominican Republic' 'Ecuador' 'Egypt' 'El Salvador' 'Estonia' 'Eswatini'
 'Ethiopia' 'Fiji' 'Finland' 'France' 'French Polynesia' 'Gabon' 'Gambia'
 'Georgia' 'Germany' 'Ghana' 'Greece' 'Grenada' 'Guatemala' 'Guinea'
 'Guinea-Bissau' 'Guyana' 'Haiti' 'Honduras' 'Hungary' 'Icel

In [13]:
faostat[['Item Code (FBS)', 'Item']].drop_duplicates().sort_values('Item Code (FBS)')


,Item Code (FBS),Item
0,S2731,Bovine Meat
13,S2732,Mutton & Goat Meat
26,S2733,Pigmeat
35,S2734,Poultry Meat
48,S2735,"Meat, Other"


In [42]:
faostat[['Area', 'Area Code (M49)']].drop_duplicates().sort_values('Area Code (M49)')

,Area,Area Code (M49)
0,Afghanistan,4
61,Albania,8
126,Algeria,12
191,Angola,24
256,Antigua and Barbuda,28
...,...,...
11410,Uzbekistan,860
11540,Venezuela (Bolivarian Republic of),862
9173,Samoa,882
11670,Yemen,887


### Summary of FAOSTAT Dataset
**Dataset size**
- Rows: 11865
- columns: 15

**Key feature/columns**
- `Area`: Country
- `Item`: Meet type
  - Related `Item Code (FBS)` mapping: 
    - `S231`: Bovine Meat
    - `S2732`: Mutton & Goat Meat
    - `S2733`: Pigmeat
     - `S2734`: Poultry Meat
     - `S2735`: Meat, Other
- `Year`: Measure  year
- `Value`: Consumption (kg/capita/year)

**Periodcoverage**
- Years: 2010 - 2022

**Geography coverage**
- number of countries: 190

**Missing names**
- `Notes` is the only column with `NaN` (irrelevant)

**Relevant notes**
- Some rows contain `Item = Meat, Other`: THis could be excluded for our analysis
- `Element` = "Fuud Supply quality (kg/capita/yr)" is the relevant input for training.

**Suggested actions for processing**
- Filter by `Element`
- Exclude `Item = "Meat, Other"`
- Rename key columns

## GDP Dataset

In [27]:
# First we simplify the name of the frame we will explore. 
gdp = dfs['gdp']

In [29]:
# File header structure
gdp.head(3)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,Unnamed: 69
0,Aruba,ABW,GDP per capita (current US$),NY.GDP.PCAP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,27441.529662,28440.051964,30082.127645,31096.205074,22855.932320,27200.061079,30559.533535,33984.790620,NaN,NaN
1,Africa Eastern and Southern,AFE,GDP per capita (current US$),NY.GDP.PCAP.CD,186.121835,186.941781,197.402402,225.440494,208.999748,226.876513,...,1329.807285,1520.212231,1538.901679,1493.817938,1344.103210,1522.393346,1628.318944,1568.159891,1673.841139,NaN
2,Afghanistan,AFG,GDP per capita (current US$),NY.GDP.PCAP.CD,NaN,NaN,NaN,NaN,NaN,NaN,...,522.082216,525.469771,491.337221,496.602504,510.787063,356.496214,357.261153,413.757895,NaN,NaN


In [30]:
gdp.tail(3)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,...,2016,2017,2018,2019,2020,2021,2022,2023,2024,Unnamed: 69
263,South Africa,ZAF,GDP per capita (current US$),NY.GDP.PCAP.CD,532.147504,545.657512,563.423009,604.536855,645.873376,684.621228,...,5651.205852,6618.335083,6914.178032,6533.711210,5580.603831,6843.399419,6523.410978,6022.542542,6253.371582,NaN
264,Zambia,ZMB,GDP per capita (current US$),NY.GDP.PCAP.CD,221.559849,209.693206,202.281031,203.219451,229.979246,287.425476,...,1239.085279,1483.465773,1463.899979,1258.986198,951.644317,1127.160779,1447.123101,1330.727806,1235.084665,NaN
265,Zimbabwe,ZWE,GDP per capita (current US$),NY.GDP.PCAP.CD,276.419784,279.016489,275.545608,277.005701,281.744539,294.145359,...,1407.420964,3448.086991,2271.852504,1683.913136,1730.453910,1724.387271,2040.546587,2156.034093,2656.409377,NaN


In [31]:
gdp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 70 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    object 
 1   Country Code    266 non-null    object 
 2   Indicator Name  266 non-null    object 
 3   Indicator Code  266 non-null    object 
 4   1960            151 non-null    float64
 5   1961            154 non-null    float64
 6   1962            156 non-null    float64
 7   1963            156 non-null    float64
 8   1964            156 non-null    float64
 9   1965            162 non-null    float64
 10  1966            163 non-null    float64
 11  1967            167 non-null    float64
 12  1968            168 non-null    float64
 13  1969            168 non-null    float64
 14  1970            190 non-null    float64
 15  1971            191 non-null    float64
 16  1972            191 non-null    float64
 17  1973            191 non-null    flo

In [32]:
gdp.isnull().sum()

Country Name        0
Country Code        0
Indicator Name      0
Indicator Code      0
1960              115
                 ... 
2021                8
2022                9
2023               17
2024               34
Unnamed: 69       266
Length: 70, dtype: int64

No Nulls in `Country Name`, `Country Code`, `Indicator Name`, `Indicator Code` but many Nulls from `1960` - `2024`

In [33]:
# General exploration, number of data and features
print('(rows, columns):',gdp.shape)

print('columns:', gdp.columns)

(rows, columns): (266, 70)
columns: Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022',
       '2023', '2024', 'Unnamed: 69'],
      dtype='object')


In [ ]:
# Cleaning our data:
gdp = gdp.drop(columns=['Unnamed: 69'])


In [50]:

year_cols = [col for col in gdp.columns if col.isdigit()]  # columns in years
gdp = gdp.dropna(subset=year_cols, how='all')

In [53]:
# General exploration, number of data and features
print('(rows, columns):',gdp.shape)

print('columns:', gdp.columns)

(rows, columns): (262, 69)
columns: Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022',
       '2023', '2024'],
      dtype='object')


In [54]:
gdp.isnull().sum()

Country Name        0
Country Code        0
Indicator Name      0
Indicator Code      0
1960              111
                 ... 
2020                4
2021                4
2022                5
2023               13
2024               30
Length: 69, dtype: int64

In [ ]:
# General exploration, number of data and features
print('Unic values in  columns')
print(faostat['Domain'].unique())
print(faostat['Item Code (FBS)'].unique())
print(faostat['Item'].unique())
print('Countries:', faostat['Area'].nunique())
print('Year range:',faostat['Year'].min(), '-' , faostat['Year'].max())
print('Suspicious values',faostat['Flag'].value_counts())

In [ ]:

print('Number of countries:', gdp['Country Name'].nunique())
print('Countries:', gdp['Country Name'].unique())


Number of countries: 266
Countries: ['Aruba' 'Africa Eastern and Southern' 'Afghanistan'
 'Africa Western and Central' 'Angola' 'Albania' 'Andorra' 'Arab World'
 'United Arab Emirates' 'Argentina' 'Armenia' 'American Samoa'
 'Antigua and Barbuda' 'Australia' 'Austria' 'Azerbaijan' 'Burundi'
 'Belgium' 'Benin' 'Burkina Faso' 'Bangladesh' 'Bulgaria' 'Bahrain'
 'Bahamas, The' 'Bosnia and Herzegovina' 'Belarus' 'Belize' 'Bermuda'
 'Bolivia' 'Brazil' 'Barbados' 'Brunei Darussalam' 'Bhutan' 'Botswana'
 'Central African Republic' 'Canada' 'Central Europe and the Baltics'
 'Switzerland' 'Channel Islands' 'Chile' 'China' "Cote d'Ivoire"
 'Cameroon' 'Congo, Dem. Rep.' 'Congo, Rep.' 'Colombia' 'Comoros'
 'Cabo Verde' 'Costa Rica' 'Caribbean small states' 'Cuba' 'Curacao'
 'Cayman Islands' 'Cyprus' 'Czechia' 'Germany' 'Djibouti' 'Dominica'
 'Denmark' 'Dominican Republic' 'Algeria'
 'East Asia & Pacific (excluding high income)'
 'Early-demographic dividend' 'East Asia & Pacific'
 'Europe & Central 

## Urban Dataset

## Education Dataset

## Enviroment Dataset

## Meat production Dataset